src/lib/ConfigCache.js

```diff
@@ -0,0 +1,104 @@
+/* ConfigCache.js
+ * 
+ * copyright (c) 2010-2016, Christian Mayer and the CometVisu contributers.
+ * 
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA
+ */
+
+/**
+ * Handles caches for cometvisu configs
+ *
+ * @author Tobias Bräutigam
+ * @since 0.10.0
+ */
+
+/* TODO:
+ - cache icons
+ */
+define([], function() {
+  var instance;
+
+  function ConfigCache() {
+    this._cacheKey = "data";
+    this._parseCacheData = null;
+    this._valid = null;
+
+    this.dump = function(xml) {
+      this.save(this._cacheKey, {
+        hash: this.toHash(xml),
+        data: templateEngine.widgetData,
+        addresses: templateEngine.ga_list,
+        configSettings: templateEngine.configSettings
+      });
+      localStorage.setItem(templateEngine.configSuffix+".body", $('body').html());
+    };
+
+    this.save = function(key, data) {
+      localStorage.setItem(templateEngine.configSuffix+"."+key, JSON.stringify(data));
+    };
+
+    this.getBody = function() {
+      return localStorage.getItem(templateEngine.configSuffix + ".body");
+    };
+
+    this.getData = function(key) {
+      if (!this._parseCacheData) {
+        this._parseCacheData = JSON.parse(localStorage.getItem(templateEngine.configSuffix + "." + this._cacheKey));
+      }
+      if (!this._parseCacheData) {
+        return null;
+      }
+      if (key) {
+        return this._parseCacheData[key];
+      } else {
+        return this._parseCacheData;
+      }
+    };
+
+    /**
+     * Returns true if there is an existing cache for the current config file
+     */
+    this.isCached = function() {
+      return localStorage.getItem(templateEngine.configSuffix + "." + this._cacheKey) !== null;
+    };
+
+    this.isValid = function(xml) {
+      // cache the result, as the config stays the same until next reload
+      if (this._valid === null) {
+        console.log("Current hash: '%s', cached hash: '%s'", this.toHash(xml), this.getData("hash"));
```

Comment:
Shall this be kept in the next release or is it just for debug?

When it's the later, please give it a comment so that everybody knows that this line can be deleted before the next release